sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -327,8 +332,25 @@ DataSource buildDatasource() throws Exception{
         if (getConnectionProperties() != null && getConnectionProperties().get() != null) {
           basicDataSource.setConnectionProperties(getConnectionProperties().get());
         }
-        return basicDataSource;
+        current = basicDataSource;
       }
+
+      // wrapping the datasource as a pooling datasource
+      DataSourceConnectionFactory connectionFactory = new DataSourceConnectionFactory(current);
+      PoolableConnectionFactory poolableConnectionFactory =
+              new PoolableConnectionFactory(connectionFactory, null);
+      GenericObjectPoolConfig poolConfig = new GenericObjectPoolConfig();
+      poolConfig.setMaxTotal(1);
+      poolConfig.setMinIdle(0);
+      poolConfig.setMinEvictableIdleTimeMillis(10000);
```

Comment:
Thanks, this looks reasonable, though is there any way to manually verify that this is working as intended in practice?

Reply:
Basically, you mean trying to implement a test to illustrate/test ?

Reply:
I don't know if it's worth it to verify the behavior automatically, I'm just asking to manually run a streaming pipeline with JdbcIO and confirm that it actually creates and releases connections when the pipeline is idle for some time (maybe enable logging in dbcp somewhere. Or at least confirm that the pipeline works at all for a few minutes).

Reply:
Ah ok, so manual test. Let me prepare a sample about that. Thanks !